base/libdl.jl

```diff
@@ -334,7 +334,7 @@ struct LazyLibraryPath
     LazyLibraryPath(pieces::Vector) = new(pieces)
 end
 LazyLibraryPath(args...) = LazyLibraryPath(collect(args))
-Base.string(llp::LazyLibraryPath) = joinpath(string.(llp.pieces)...)::String
+Base.string(llp::LazyLibraryPath) = joinpath(String[string(p) for p in llp.pieces])
```

Comment:
Arguably prettier, ignore if you don't agree:

```suggestion
Base.string(llp::LazyLibraryPath) = joinpath(collect(String, Iterators.map(string, llp.pieces)))
```